### ISSUE TITLE
404 response to valid route

### ISSUE THREAD

**Issue by yankeeinlondon** [traefik/traefik#9235]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

![image](https://user-images.githubusercontent.com/980757/183488580-304a0441-d52a-4753-bf7c-9c442ec5e099.png)

I am using docker-compose and have my **traefik** service setup as:

```yml
traefik:
    image: traefik:latest
    container_name: traefik
    restart: unless-stopped
    command: --api.insecure=true --providers.docker
    security_opt:
      - no-new-privileges:true
    networks:
      - web
    ports:
      - 80:80
      - 443:443
      - 8080:8080
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /home/ken/data/traefik:/etc/traefik
    labels:
      - traefik.enable=true
```

I also have the classic example of the **whoami** service:

```yml
  whoami:
    image: traefik/whoami
    container_name: whoami
    restart: unless_stopped
    labels:
      - traefik.enable=true
      - traefik.http.routers.whoami.rule=Host(`whoami.local`)
      - traefik.http.routers.whoami.entrypoints=websecure
      - traefik.http.services.whoami.loadbalancer.server.port=80
    networks:
      - web
```

### What did you see instead?

Looking at the dashboard everything looks fine but when I actually point a browser at `whoami.local` (or use **curl** to do the same) I get a 404 error. However, if I were to actually run curl against the docker service -- `curl http://172.23.0.2:80` -- it does indeed return whoami info.

![image](https://user-images.githubusercontent.com/980757/183489973-43b463fb-52f1-4bea-ad0e-ecb47c7e77ce.png)

Note: I've tried adding some other routes to services like **heimdall**, etc. but they all end up with the same problem. I am pulling my hair out ... i have spent three days trying to just get a basic setup setup. :( 

### What version of Traefik are you using?

2.8.1

## `traefik.yml`

```yml
api:
  dashboard: true
  debug: true
  insecure: true
log:
  level: DEBUG

entryPoints:
  web:
    address: ":80"
    http:
      redirections:
        entryPoint:
          to: websecure
          scheme: https

  websecure:
    address: ":443"

serversTransport:
  insecureSkipVerify: true

providers:
  docker:
    endpoint: "unix:///var/run/docker.sock"
    exposedByDefault: false
    network: web
    watch: true
```

**Comment by yankeeinlondon** [comment]
if it helps, here's another screenshot showing all the routes setup:
![image](https://user-images.githubusercontent.com/980757/183491580-d0c64cc7-a52b-400d-8ba6-18cb4b581f51.png)

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

We dedicate the issue tracker to bug reports and feature requests only. My advanced AI has spotted that your issue might be a configuration problem or relates to something that doesn't look like a bug.

To confirm this, please join our [Community Forum](https://community.containo.us/) and reach out to us on the [Traefik section](https://community.containo.us/c/traefik).

In case I'm wrong (well, that would be embarrassing :sweat_smile:), my developers will re-open the issue and fix me!

In the meantime, you can double check [Traefik's documentation](https://docs.traefik.io/).

**Comment by yankeeinlondon** [comment]
WTF! This certainly looks like bug.

**Comment by yankeeinlondon** [comment]
Infuriating!

**Comment by ldez** [comment]
Hello,

I think that you are looking for this:

> ### The Static Configuration
>
> There are three different, **mutually exclusive** (i.e. you can use only one at the same time), ways to define static configuration options in Traefik:
>
> - In a configuration file
> - In the command-line arguments
> - As environment variables

https://doc.traefik.io/traefik/getting-started/configuration-overview/#the-static-configuration

If you need more help, please come to the [forum](https://community.traefik.io/)

**Comment by yankeeinlondon** [comment]
No I'm fully aware of that. I am using a file for static config and it picks this up. I am also using dynamic configuration via labels and it picks that up too. It just doesn't resolve any routes to anything but a 404 or some unclear error on redirection. I am looking desperately for some clue in example code and logs but nothing seems to work.

**Comment by ldez** [comment]
The configuration you provide and the docker-compose cannot work, because there are several problems:
- you are use CLI flags (`command: --api.insecure=true --providers.docker`) and file (`- /home/ken/data/traefik:/etc/traefik`) for the static configuration.
- the TLS is missing of `websecure`
- etc.

I will help you on the forum, if you don't want, the answer is:

```
version: "3.9"

services:
  traefik:
    image: traefik:v2.8
    container_name: traefik
    command:
      - --api.insecure=true
      - --log.level=DEBUG
      - --providers.docker
      - --entrypoints.web.address=:80
      - --entrypoints.web.http.redirections.entryPoint.to=websecure
      - --entrypoints.web.http.redirections.entrypoint.scheme=https
      - --entrypoints.websecure.address=:443
      - --entrypoints.websecure.http.tls=true
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro

  whoami:
    image: traefik/whoami:v1.8.1
    container_name: whoami
    labels:
      traefik.http.routers.whoami.rule: Host(`whoami.localhost`)
      traefik.http.routers.whoami.entrypoints: websecure
```

**Comment by yankeeinlondon** [comment]
@ldez thank  you for taking the time ... i have "wanted" to use Traefik for a long while but both times I tried to start (on K8s and Docker) it was VERY painful. I guess once the basics are understood the documentation may serve as a good referrence but there is not enough "starter content" in my opinion. Anyway, thank you and I learned a lot just by looking at this example's syntax in the command and labels section.
